xonsh/prompt/vc.py

```diff
@@ -175,19 +164,31 @@ def current_branch():
 
 
 def _git_dirty_working_directory(q, include_untracked):
-    status = None
     denv = builtins.__xonsh__.env.detype()
     try:
-        cmd = ["git", "status", "--porcelain"]
+        # Borrowed from this conversation
+        # https://github.com/sindresorhus/pure/issues/115
         if include_untracked:
-            cmd.append("--untracked-files=normal")
+            cmd = [
+                "git",
+                "status",
+                "--porcelain",
+                "--quiet",
```

Comment:
I don't think this is correct, nor is it it correct in current `master`:

`git status` does not have a `--quiet` option; it does not exist in the man page and errors with

```
error: unknown option `quiet'
```

I also don't think that just calling `bool()` below on the return code is a good idea; it would be better to check the exact exit codes that are expected. That would probably have revealed this problem.

Reply:
ahh nice catches! you're right, using "bool" hid the error. thank you! i'll try to diff in a fix soon, unless you'd prefer to take this on